cpp/src/structs/utilities.cpp

```diff
@@ -215,54 +215,6 @@ flattened_table flatten_nested_columns(table_view const& input,
 
 namespace {
 
-/**
- * @brief Dispatcher to conservatively check if the input column may contain any non-empty nulls.
- *
- * For performance reason, no non-empty null will actually be checked. Instead, this will perform
- * conservative checking to see if the input column contains any lists or strings column that is
- * nulllable. As such, it may return false positive answer.
- */
-struct may_contain_non_empty_nulls_dispatch {
-  template <typename T>
-  bool operator()(column_view const& col) const
-  {
-    return std::any_of(col.child_begin(), col.child_end(), [](auto const& child) {
-      return type_dispatcher(child.type(), may_contain_non_empty_nulls_dispatch{}, child);
-    });
-  }
-};
-
-template <>
-bool may_contain_non_empty_nulls_dispatch::operator()<cudf::string_view>(
-  column_view const& col) const
-{
-  return col.nullable();
-}
-
-template <>
-bool may_contain_non_empty_nulls_dispatch::operator()<cudf::list_view>(column_view const& col) const
-{
-  // A lists column may have empty data (i.e., having child column of type EMPTY).
-  // Thus, it needs to be sanitized only if the child column is not empty.
-  return col.nullable() &&
-         col.child(lists_column_view::child_column_index).type().id() != type_id::EMPTY;
-}
-
-/**
- * @brief Conservatively check if the input column may contain any non-empty nulls.
- *
- * This will just call `type_dispatcher` on the `may_contain_non_empty_nulls_dispatch` functor.
- *
- * @param col The input column to check
- */
-bool may_contain_non_empty_nulls(column_view const& col)
-{
-  // Type EMPTY is not handled by `type_dispatcher`.
-  if (col.type().id() == type_id::EMPTY) { return false; }
-
-  return type_dispatcher(col.type(), may_contain_non_empty_nulls_dispatch{}, col);
```

Comment:
Looks like it will be possible to remove `#include <cudf/utilities/type_dispatcher.hpp>`
Perhaps also check some of these includes as well?
```
#include <cudf/detail/copy.hpp>
#include <cudf/lists/list_view.hpp>
#include <cudf/strings/string_view.cuh>
#include <rmm/device_buffer.hpp>
#include <bitset>
```
(these may be wrong and/or there may be more)